File: app/src/main/java/com/romainpiel/darktown/MarkDownEditText.java
package com.romainpiel.darktown;

import android.content.Context;
import android.os.Parcelable;
import android.util.AttributeSet;
import android.widget.EditText;

public class MarkDownEditText extends EditText {

    private final MarkDownTextWatcher textWatcher;

    public MarkDownEditText(Context context) {
        super(context);
        textWatcher = new MarkDownTextWatcher();
        initWatcher();
    }

    public MarkDownEditText(Context context, AttributeSet attrs) {
        super(context, attrs);
        textWatcher = new MarkDownTextWatcher();
        initWatcher();
    }

    public MarkDownEditText(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        textWatcher = new MarkDownTextWatcher();
        initWatcher();
    }

    private void initWatcher() {
        removeTextChangedListener(textWatcher);
        addTextChangedListener(textWatcher);
    }

    @Override
    public void onRestoreInstanceState(Parcelable state) {
        super.onRestoreInstanceState(state);
        initWatcher();
    }
}

File: app/src/main/kotlin/com/romainpiel/darktown/MarkDownEditText.kt
package com.romainpiel.darktown

import android.content.Context
import android.os.Parcelable
import android.util.AttributeSet
import android.widget.EditText

class MarkDownEditText : EditText {

    private val textWatcher: MarkDownTextWatcher

    constructor(context: Context) : super(context) {
        textWatcher = MarkDownTextWatcher()
        initWatcher()
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        textWatcher = MarkDownTextWatcher()
        initWatcher()
    }

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        textWatcher = MarkDownTextWatcher()
        initWatcher()
    }

    private fun initWatcher() {
        removeTextChangedListener(textWatcher)
        addTextChangedListener(textWatcher)
    }

    override fun onRestoreInstanceState(state: Parcelable) {
        super.onRestoreInstanceState(state)
        initWatcher()
    }
}
